src/AutoRest.CSharp/Common/Output/Builders/JsonSerializationMethodsBuilder.cs

```diff
@@ -195,6 +210,64 @@ private static MethodBodyStatement[] WriteObject(JsonObjectSerialization seriali
                 utf8JsonWriter.WriteEndObject()
             };
 
+        private static MethodBodyStatement[] BuildJsonModelWriteMethodBody(Method jsonModelWriteCoreMethod, Utf8JsonWriterExpression utf8JsonWriter)
+        {
+            var coreMethodSignature = jsonModelWriteCoreMethod.Signature;
+
+            return new[]
+            {
+                utf8JsonWriter.WriteStartObject(),
+                This.Invoke(coreMethodSignature.Name, coreMethodSignature.Parameters.Select(p => (ValueExpression)p).ToList()).ToStatement(),
+                utf8JsonWriter.WriteEndObject(),
+            };
+        }
+
+        // void JsonModelWriteCore(Utf8JsonWriter writer, ModelReaderWriterOptions options)
+        private static Method BuildJsonModelWriteCoreMethod(JsonObjectSerialization serialization, CSharpType? iPersistableModelTInterface, bool hasInherits, bool isSealed)
+        {
+            MethodSignatureModifiers modifiers = hasInherits ? MethodSignatureModifiers.Protected | MethodSignatureModifiers.Override : MethodSignatureModifiers.Protected | MethodSignatureModifiers.Virtual;
+            if (serialization.Type.IsValueType || isSealed)
+                modifiers = MethodSignatureModifiers.Private;
+
+            var writerParameter = new Parameter("writer", $"The JSON writer.", typeof(Utf8JsonWriter), null, ValidationType.None, null);
+            var optionsParameter = new Parameter("options", $"The client options for reading and writing models.", typeof(ModelReaderWriterOptions), null, ValidationType.None, null);
```

Comment:
I think the knownparameters.serializations type also defines parameters for writer and options, why we are not using them?

Reply:
That one doesn't have description, therefore the generate API won't have description.

Reply:
Then add parameters with descriptions in the knownparameters and use them here. Also below we should use these new parameters